tests/ManiaTemplates.Tests/Components/MtComponentScriptTest.cs

```diff
@@ -0,0 +1,63 @@
+﻿using System.Xml;
+using ManiaTemplates.Components;
+
+namespace ManiaTemplates.Tests.Components;
+
+public class MtComponentScriptTest
+{
+    private readonly ManiaTemplateEngine _templateEngine = new();
+
+    public MtComponentScriptTest()
+    {
+        _templateEngine.AddManiaScriptFromString("res", "text");
+    }
+
+    [Theory]
+    [InlineData("<script>text</script>", false, false)]
+    [InlineData("<script main=''>text</script>", true, false)]
+    [InlineData("<script once=''>text</script>", false, true)]
+    [InlineData("<script main='' once=''>text</script>", true, true)]
+    [InlineData("<script resource='res'/>", false, false)]
+    [InlineData("<script resource='res' main=''/>", true, false)]
+    [InlineData("<script resource='res' once=''/>", false, true)]
+    [InlineData("<script resource='res' main='' once=''/>", true, true)]
+    public void ShouldLoadManiaScript(string input, bool expectedMain, bool expectedOnce)
+    {
+        var document = new XmlDocument();
+        document.LoadXml(input);
+
+        var res = MtComponentScript.FromNode(_templateEngine, document.DocumentElement!);
+        Assert.Equal("text", res.Content);
+        Assert.Equal(expectedMain, res.Main);
+        Assert.Equal(expectedOnce, res.Once);
+    }
+
+    [Theory]
+    [InlineData("<script/>")]
+    [InlineData("<script main='false' once='true' />")]
+    public void ShouldFailToLoadEmptyManiaScript(string input)
+    {
+        var document = new XmlDocument();
+        document.LoadXml(input);
+
+        var exception = Assert.Throws<Exception>(() => MtComponentScript.FromNode(_templateEngine, document.DocumentElement!));

```

Comment:
Generic exception should not be used as it makes it more difficult programmatically to differentiate what happened. It also makes tests cleaner as you don't have to compare the exception message. The Assert.Throws will do everything for you.